plugins/wallabag.koplugin/main.lua

```diff
@@ -1096,172 +1390,234 @@ Restart KOReader after editing the config file.]]), BD.dirpath(DataStorage:getSe
-        download_original_document = self.download_original_document,
-        download_queue        = self.download_queue,
+        server_url                    = self.server_url,
+        client_id                     = self.client_id,
+        client_secret                 = self.client_secret,
+        username                      = self.username,
+        password                      = self.password,
+        directory                     = self.directory,
+        archive_directory             = self.archive_directory,
+        archive_abandoned             = self.archive_abandoned,
+        archive_finished              = self.archive_finished,
+        archive_read                  = self.archive_read,
+        articles_per_sync             = self.articles_per_sync,
+        auto_archive                  = self.auto_archive,
+        auto_tags                     = self.auto_tags,
+        delete_instead                = self.delete_instead,
+        download_original_document    = self.download_original_document,
+        filter_tag                    = self.filter_tag,
+        ignore_tags                   = self.ignore_tags,
+        remove_abandoned_from_history = self.remove_abandoned_from_history,
+        remove_finished_from_history  = self.remove_finished_from_history,
+        remove_read_from_history      = self.remove_read_from_history,
+        send_review_as_tags           = self.send_review_as_tags,
+        sync_remote_archive           = self.sync_remote_archive,
+        upload_queue                  = self.upload_queue,
+        use_local_archive             = self.use_local_archive,
     }
+
     self.wb_settings:saveSetting("wallabag", tempsettings)
     self.wb_settings:flush()
-
-end
+end -- Wallabag:saveSettings
 
 function Wallabag:readSettings()
-    local wb_settings = LuaSettings:open(DataStorage:getSettingsDir().."/wallabag.lua")
+    local wb_settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/wallabag.lua")
     wb_settings:readSetting("wallabag", {})
     return wb_settings
-end
-
-function Wallabag:saveWBSettings(setting)
-    if not self.wb_settings then self.wb_settings = self:readSettings() end
-    self.wb_settings:saveSetting("wallabag", setting)
-    self.wb_settings:flush()
-end
+end -- Wallabag:readSettings
 
+--- Handler for addWallabagArticle event.
+--- Uploads a new article to wallabag directly if there is a network connection, or add it to the
+-- local upload queue.
 function Wallabag:onAddWallabagArticle(article_url)
     if not NetworkMgr:isOnline() then
-        self:addToDownloadQueue(article_url)
+        self:addToUploadQueue(article_url)
         UIManager:show(InfoMessage:new{
-            text = T(_("Article added to download queue:\n%1"), BD.url(article_url)),
-            timeout = 1,
-         })
+            text = T(_("Article added to upload queue:\n%1"), BD.url(article_url)),
+            timeout = 10,
```

Comment:
10 seems a bit excessive. I wouldn't put it at more than 3 myself.

Also I'm not quite convinced by changing download to upload here. That seems developer-centric rather than user-centric.